Sources/SwiftKuery/ResultFetcher.swift

```diff
@@ -33,4 +33,8 @@ public protocol ResultFetcher {
     ///
     /// - Returns: An array of column titles of type String.
     func fetchTitles() -> [String]
+
```

Comment:
Do we need an async `fetchTitles()`?

Reply:
Currently none of the ResultFetcher methods are asynchronous. Any changes would need to be exposed via ResultSet. It might make more sense to revisit this once we have proper non-blocking IO to the database as limitations imposed at the plugin layer may make this harder than it sounds (no concurrent use of connections). Would a follow up work item suffice?

Reply:
Sounds fair enough.

Reply:
`func fetchNext(callback: ([Any?]?) ->())` is async though?

Reply:
The API appears to be but it's implementation in our plugins (MySQL, PostgreSQL and SQLite) is not. I'd suggest that if we want to make ResultSet an asynchronous API we do it under a different issue after we have merged these changes, but preferably before we tag the next major.

Reply:
https://github.com/IBM-Swift/Swift-Kuery/issues/160 opened to address